ghproxy/ghcache/coalesce.go

```diff
@@ -129,7 +129,7 @@ func (r *requestCoalescer) RoundTrip(req *http.Request) (*http.Response, error)
 		return resp, nil
 	}()
 
-	ghmetrics.CollectCacheRequestMetrics(string(cacheMode), req.URL.Path)
+	ghmetrics.CollectCacheRequestMetrics(string(cacheMode), req.URL.Path, req.Header.Get("User-Agent"))
```

Comment:
Should we strip out the version and just use the component name to avoid creating too many prometheus label values? I don't know what the scale limits of prometheus are in terms of number of label values, but I know that was a concern with the request path metrics.

Reply:
Hm, good question. Since we generally don't expect to have separate versions of e.g. `hook` running at once it may be fine to do. Over a four week span I expect to see ~80 (2x/day * 5day/week * 4 weeks) distinct versions of a component, so the cardinality is in the hundreds for all components -- generally I believe that to be fine, it's when we go into thousands and up that we see performance impacts.